Src/FluentAssertions/Xml/XDocumentAssertions.cs

```diff
@@ -344,7 +344,201 @@ public AndWhichConstraint<XDocumentAssertions, IEnumerable<XElement>> HaveElemen
+    /// </summary>
+    /// <param name="expectedElement">
+    /// The name of the expected child element of the current element's <see cref="XDocument"/>.
+    /// </param>
+    /// <param name="expectedValue">
+    /// The expected value of this particular element.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndWhichConstraint<XDocumentAssertions, XElement> HaveElementWithValue(string expectedElement,
+        string expectedValue, [StringSyntax("CompositeFormat")] string because = "", params object[] becauseArgs)
+    {
+        Guard.ThrowIfArgumentIsNull(expectedElement, nameof(expectedElement));
+        Guard.ThrowIfArgumentIsNull(expectedValue, nameof(expectedValue));
+
+        return HaveElementWithValue(XNamespace.None + expectedElement, expectedValue, because, becauseArgs);
+    }
+
+    /// <summary>
+    /// Asserts that the <see cref="XDocument.Root"/> of the current <see cref="XDocument"/> has the specified child element
+    /// with the specified <paramref name="expectedValue"/> name.
+    /// </summary>
+    /// <param name="expectedElement">
+    /// The full name <see cref="XName"/> of the expected child element of the current element's <see cref="XDocument"/>.
+    /// </param>
+    /// <param name="expectedValue">
+    /// The expected value of this particular element.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndWhichConstraint<XDocumentAssertions, XElement> HaveElementWithValue(XName expectedElement,
+        string expectedValue, [StringSyntax("CompositeFormat")] string because = "", params object[] becauseArgs)
+    {
+        Guard.ThrowIfArgumentIsNull(expectedElement, nameof(expectedElement));
+        Guard.ThrowIfArgumentIsNull(expectedValue, nameof(expectedValue));
+
+        assertionChain
+            .ForCondition(Subject is not null)
+            .BecauseOf(because, becauseArgs)
+            .FailWith(
+                "Expected {context:subject} to have an element {0} with value {1}{reason}, but the element itself is <null>.",
+                expectedElement.ToString(), expectedValue);
+
+        IEnumerable<XElement> xElements = [];
+
+        if (assertionChain.Succeeded)
```

Comment:
The new `AssertionChain` construct allows to write this like that:

```csharp
assertionChain
    .WithExpectation("Expected {context:subject} to have an element {0} with value {1}{reason}, ",
        expectedElement.ToString(), expectedValue, chain => chain
            .ForCondition(Subject is not null)
            .BecauseOf(because, becauseArgs)
            .FailWith(
                "but the element itself is <null>.",
                expectedElement.ToString(), expectedValue))
    .Then
    .WithExpectation("Expected {context:subject} to have an element {0} with value {1}{reason}, ",
        expectedElement, expectedValue, chain => chain
            .BecauseOf(because, becauseArgs)
            .Given(() =>
            {
                xElements = Subject!.Root!.Elements(expectedElement).ToList();

                return xElements;
            })
            .ForCondition(collection => collection.Any())
            .FailWith("but the element {0} isn't found.", expectedElement)
            .Then
            .ForCondition(collection => collection.Any(e => e.Value == expectedValue))
            .FailWith("but the element {0} does not have such a value.", expectedElement));
```
@dennisdoomen Should I go for that (and everywhere else in this PR)?

Reply:
Totally up to you. I'm fine merging it like this.

Reply:
Me too.. I like such chainings :)